src/modules/px4iofirmware/controls.c

```diff
@@ -93,6 +94,7 @@ bool dsm_port_input(uint16_t *rssi, bool *dsm_updated, bool *st24_updated, bool
 		r_raw_rc_flags &= ~(PX4IO_P_RAW_RC_FLAGS_FRAME_DROP);
 		r_raw_rc_flags &= ~(PX4IO_P_RAW_RC_FLAGS_FAILSAFE);
 
+		*rssi = spektrum_rssi;
```

Comment:
You can ensure that the RSSI arrives on the FMU side like this:

```
/* ensure ADC RSSI is disabled */
r_setup_features &= ~(PX4IO_P_SETUP_FEATURES_ADC_RSSI);
```

Reply:
Thanks for sending me on the right track here. After this, there's two other things to do in order for digital rssi to work. These changes aren't unique to spektrum, and digital rssi is broken on other platforms without them too.

1. set storage for `uint16_t rssi` in `controls_tick()` to static, or store `rssi` in some other fashion. `controls_tick()` calls `dsm_port_input()` multiple times per rc frame, so unless tick remembers the valid rssi value until it comes in again, it will be reset to zero on the subsequent tick.

2. `src/drivers/px4io/px4io.cpp:1772`: read the value of `regs[PX4IO_P_RAW_RC_NRSSI]` into `input_rc.rssi`. This value is set to 255 unless there's an ADC RSSI coming in. There's a comment that reads "we do not actually get digital RSSI regs[PX4IO_P_RAW_RC_NRSSI]" but this doesn't seem to be the case.